File: java-spring/src/main/java/org/burgas/javaspring/config/RedisConfig.java
package org.burgas.javaspring.config;

import org.burgas.javaspring.dto.identity.IdentityFullResponse;
import org.burgas.javaspring.dto.note.NoteFullResponse;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.redis.connection.RedisConnectionFactory;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.data.redis.serializer.JacksonJsonRedisSerializer;
import org.springframework.data.redis.serializer.StringRedisSerializer;

@Configuration
public class RedisConfig {

    @Bean
    public RedisTemplate<String, IdentityFullResponse> identityRedisTemplate(RedisConnectionFactory connectionFactory) {
        RedisTemplate<String, IdentityFullResponse> template = new RedisTemplate<>();
        template.setConnectionFactory(connectionFactory);
        template.setKeySerializer(new StringRedisSerializer());
        template.setValueSerializer(new JacksonJsonRedisSerializer<>(IdentityFullResponse.class));
        return template;
    }

    @Bean
    public RedisTemplate<String, NoteFullResponse> noteRedisTemplate(RedisConnectionFactory connectionFactory) {
        RedisTemplate<String, NoteFullResponse> template = new RedisTemplate<>();
        template.setConnectionFactory(connectionFactory);
        template.setKeySerializer(new StringRedisSerializer());
        template.setValueSerializer(new JacksonJsonRedisSerializer<>(NoteFullResponse.class));
        return template;
    }
}

File: kotlin-spring/src/main/kotlin/org/burgas/kotlinspring/config/RedisConfig.kt
package org.burgas.kotlinspring.config

import org.burgas.kotlinspring.entity.identity.IdentityFullResponse
import org.burgas.kotlinspring.entity.note.NoteFullResponse
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.redis.connection.RedisConnectionFactory
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.data.redis.serializer.JacksonJsonRedisSerializer
import org.springframework.data.redis.serializer.StringRedisSerializer
import tools.jackson.databind.ObjectMapper

@Configuration
class RedisConfig {

    @Bean
    fun identityRedisTemplate(connectionFactory: RedisConnectionFactory): RedisTemplate<String, IdentityFullResponse> {
        val template = RedisTemplate<String, IdentityFullResponse>()
        template.connectionFactory = connectionFactory

        val mapper = ObjectMapper()

        val serializer = JacksonJsonRedisSerializer(mapper, IdentityFullResponse::class.java)

        template.keySerializer = StringRedisSerializer()
        template.valueSerializer = serializer
        template.hashKeySerializer = StringRedisSerializer()
        template.hashValueSerializer = serializer

        return template
    }

    @Bean
    fun noteRedisTemplate(connectionFactory: RedisConnectionFactory): RedisTemplate<String, NoteFullResponse> {
        val template = RedisTemplate<String, NoteFullResponse>()
        template.connectionFactory = connectionFactory

        val mapper = ObjectMapper()

        val serializer = JacksonJsonRedisSerializer(mapper, NoteFullResponse::class.java)

        template.keySerializer = StringRedisSerializer()
        template.valueSerializer = serializer
        template.hashKeySerializer = StringRedisSerializer()
        template.hashValueSerializer = serializer

        return template
    }
}